benchmark/key.cpp

```diff
@@ -0,0 +1,99 @@
+#include <fc/crypto/public_key.hpp>
+#include <fc/crypto/private_key.hpp>
+#include <fc/crypto/signature.hpp>
+#include <fc/utility.hpp>
+
+#include <benchmark.hpp>
+
+using namespace fc::crypto;
+using namespace fc;
+using namespace std::literals;
+
+namespace benchmark {
+
+void k1_benchmarking() {
+   auto payload = "Test Cases";
+   auto digest = sha256::hash(payload, const_strlen(payload));
+   auto key = private_key::generate<ecc::private_key_shim>();
+
+   auto sign_f = [&]() {
+      key.sign(digest);
```

Comment:
I would recommend disabling the canonical requirement for the signing. Otherwise the value could swing widely based on the private key you happened to randomly generate.

Reply:
Good point! I will change to use fixed private keys so we can compare the values consistently across different runs.

Reply:
if you just pass false here for the second parameter, it'll be constant time for all private keys anyways

Reply:
Thanks! I am benchmarking both canonical and non-canonical cases.

Reply:
imo there is very little utility in the canonical case -- I would never look at it personally. It's dependent on both the private key and payload, both of which are fixed here, thus making the time spent always be some constant multiplied with the non-canonical time. Up to you though.

Reply:
I will remove the canonical case.